### PR TITLE
Avaliação - Auro

### PR DESCRIPTION
**Pull request by auromota**
Thanks, Edy!

### CODE REVIEW

File: app/studentsApi.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        if (grade < 60) {
+            return 'Final';
+        }
+        return 'Approved';
+    }
+
+    function calculateGrade(student) {
+        if (student.grades.length) {
+            return student.grades.reduce((previous, current) => previous + current, 0) / student.grades.length;
```
Comment (edysegura) [inline app/studentsApi.js:38]: If we use only numbers we can omit the initial values:

image

Comment (edysegura) [reply app/studentsApi.js:38]: It is a nice case to use reduce here! ;)

File: public/js/studentController.js [hunk 2]
```diff
@@ -1,20 +1,41 @@
...
     function getStudentById(id) {
         $.ajax({
-            method: "GET",
-            url: "/api/students/" + id
-        })
-        .done(student => {
-            console.log(student)
-        });
+            method: 'GET',
+            url: '/students?mat=' + id
```
Comment (edysegura) [inline public/js/studentController.js:8]: Why not to use REST API pattern "/api/students/2017001"? 
Look at "Provide Sensible Resource Names" into [this tips](http://www.restapitutorial.com/lessons/restquicktips.html)

**Review by edysegura — APPROVED** [review]
Pretty good! I have only few comments for you.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+let app = express();
+
+app.use(bodyParser.json());
+app.use(express.static('public'));
+
+router(app);
+
+app.listen(8080, () => {
+    console.log('Server on-line!');
```
Comment (edysegura) [inline index.js:15]: It is good to know in console which port the application is running!

File: app/studentsApi.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    function getStudent(req, res, next) {
+        let id = req.query.mat;
+        if (id) {
+            let student = students.find(student => student.id === id);
+            student.status = getStatus(student);
+            if (student) {
+                res.status(200).send(student);
+            } else {
+                res.status(200).send({ message: 'Student not found.' });
```
Comment (edysegura) [inline app/studentsApi.js:18]: When the API is working but we didn't return anything we can use 204 [http status code](http://www.restapitutorial.com/httpstatuscodes.html)

**Review by edysegura — COMMENTED** [review]
Just a couple of more comments

**Comment by auromota** [comment]
Hey, @edysegura! I haven't been in GitHub for a while. Just saw your comments now. Thanks for the tips. 👍

**Pull request closed without merging**